test/libdnf/transaction/test_workflow.cpp

```diff
@@ -0,0 +1,148 @@
+/*
+Copyright (C) 2017-2020 Red Hat, Inc.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+
+#include "test_workflow.hpp"
+
+#include "libdnf/common/sack/query_cmp.hpp"
+#include "libdnf/transaction/transaction.hpp"
+
+#include <string>
+
+
+using namespace libdnf::transaction;
+
+
+CPPUNIT_TEST_SUITE_REGISTRATION(TransactionWorkflowTest);
+
+
+void TransactionWorkflowTest::test_default_workflow() {
+    auto base = new_base();
+
+    // create an empty Transaction object
+    auto trans = base->get_transaction_sack().new_transaction();
+    CPPUNIT_ASSERT_EQUAL(TransactionState::UNKNOWN, trans->get_state());
+
+    // set packages used to perform the transaction
+    trans->add_runtime_package("dnf-3.5.1-1.fc29.noarch");
+
+    // set vars
+    trans->set_releasever("26");
+
+    // populate goal
+    // resolve dependencies
+    // prepare RPM transaction
+
+    // add rpm packages to the transaction
+
+    // bash-4.4.12-5.fc26.x86_64
+    auto & rpm_bash = trans->new_package();
+    rpm_bash.set_name("bash");
+    rpm_bash.set_epoch("0");
+    rpm_bash.set_version("4.4.12");
+    rpm_bash.set_release("5.fc26");
+    rpm_bash.set_arch("x86_64");
+    rpm_bash.set_repoid("base");
+    rpm_bash.set_action(TransactionItemAction::INSTALL);
```

Comment:
Data are important to save it into swdb, but are useless for runtime. rpm_bash object cannot be used for rpm_transaction.

Reply:
I take it as a note or a statement that doesn't require any change.
Or did you want to change anything?

Reply:
I think that we agreed that it will be refactored during integration with rpm transaction.